### ISSUE TITLE
[webview_flutter] iOS Media Control Notification not showing up reliably (MPNowPlayingInfoCenter)

### ISSUE THREAD

**Issue by Aulig** [flutter/flutter#118380]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Play the audio file in the webview
3. Pull down on the status bar to view the notification center.

**Expected results:** 
This is what the media control notification should look like.
![grafik](https://user-images.githubusercontent.com/32148066/212059522-258d6aa2-0e77-4e26-a7b5-8d677caa117e.png)

**Actual results:** 
Often times no media control notification shows up.
![grafik](https://user-images.githubusercontent.com/32148066/212054708-5e899294-570f-4b2c-ab46-52361af4849b.png)

The same code works sometimes and doesn't other times. It seems like it works around 1 in 8 times. Normally it should work every time of course.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(const MaterialApp(home: WebViewExample()));

class WebViewExample extends StatefulWidget {
  const WebViewExample({Key? key}) : super(key: key);

  @override
  State<WebViewExample> createState() => _WebViewExampleState();
}

class _WebViewExampleState extends State<WebViewExample> {
  late final WebViewController _controller;

  @override
  void initState() {
    super.initState();

    _controller = WebViewController()
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..loadHtmlString("""
      <!DOCTYPE html>
      <html>
        <head>
          <meta name="viewport" content="width=device-width, initial-scale=1.0">
        </head>
        <body>
          <h1>Test</h1>
          <audio controls>
            <source src="https://webtoapp.design/static/audio/wunder.mp3" type="audio/mpeg">
            Your browser does not support the audio element.
          </audio>
        </body>
      </html>
      """);
  }

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: WebViewWidget(controller: _controller),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

  Verbose logs are too large, see attached files.

[notworkingrun.txt](https://github.com/flutter/flutter/files/10401479/notworkingrun.txt)
[workingrun.txt](https://github.com/flutter/flutter/files/10401484/workingrun.txt)

```
Analyzing wvtest...                                                     
No issues found! (ran in 2.6s)
```

```
[✓] Flutter (Channel stable, 3.3.8, on macOS 12.0.1 21A559 darwin-x64, locale
    en)
    • Flutter version 3.3.8 on channel stable at
      /Users/administrator/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 52b3dc25f6 (9 weeks ago), 2022-11-09 12:09:26 +0800
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/administrator/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/administrator/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/administrator/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✗] Chrome - develop for the web (Cannot find Chrome executable at
    /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)

[✓] Connected device (2 available)
    • iPhone 8 Plus (mobile) • F557FC52-2C29-4B3E-B03C-4EC0CA8CA384 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64
      • macOS 12.0.1 21A559 darwin-x64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue on an iOS 16 simulator with the current master version (3.7.0-19.0.pre.13).

<details><summary>Code Sample (Same as OP)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(const MaterialApp(home: WebViewExample()));

class WebViewExample extends StatefulWidget {
  const WebViewExample({Key? key}) : super(key: key);

  @override
  State<WebViewExample> createState() => _WebViewExampleState();
}

class _WebViewExampleState extends State<WebViewExample> {
  late final WebViewController _controller;

  @override
  void initState() {
    super.initState();

    _controller = WebViewController()
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..loadHtmlString("""
      <!DOCTYPE html>
      <html>
        <head>
          <meta name="viewport" content="width=device-width, initial-scale=1.0">
        </head>
        <body>
          <h1>Test</h1>
          <audio controls>
            <source src="https://webtoapp.design/static/audio/wunder.mp3" type="audio/mpeg">
            Your browser does not support the audio element.
          </audio>
        </body>
      </html>
      """);
  }

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: WebViewWidget(controller: _controller),
    );
  }
}
```

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.7.0-19.0.pre.13, on macOS 13.0.1 22A400 darwin-arm64, locale en-US)
    • Flutter version 3.7.0-19.0.pre.13 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f8628b5cb1 (4 hours ago), 2023-01-11 18:48:03 -0800
    • Engine revision 4a8d6866a1
    • Dart version 3.0.0 (build 3.0.0-112.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 71.3.6
    • Dart plugin version 223.8214.16

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8214.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (4 available)
    • iPhone 14 Pro (mobile)     • A75A00DB-CD34-4A0D-BFD9-2DA753E5A5EE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • iPhone 14 Pro Max (mobile) • 6BC61E88-4DE0-4BCB-8599-DA5294269146 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Brave Browser 108.1.46.153

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by stuartmorgan** [comment]
If you make a trivial non-Flutter iOS app with a WKWebView, do you see different behavior? This seems unlikely to be something we have control over in the plugin.

**Comment by Aulig** [comment]
Thanks for your feedback Stuart - I thought exactly the same. So I tried it a couple times in this WKWebView test app: https://apps.apple.com/us/app/webview-wkwebview-and-uiwebview-rendering/id928647773 (since I have no clue about native iOS development)

It worked fine there but now upon further testing it seems like it doesn't consistently work there either. After many tests it now broke in Safari as well.

I still have no idea why it seems to break more often in the webview, but that might just be a very large coincidence.

Additionally I noticed that if you pause and resume the audio, it makes the media controls show up consistently. Very weird behaviour.
